src/otx/v2/adapters/torch/mmengine/mmseg/dataset.py

```diff
@@ -0,0 +1,252 @@
+        """
+        super().__init__(
+            task,
+            train_type,
+            train_data_roots,
+            train_ann_files,
+            val_data_roots,
+            val_ann_files,
+            test_data_roots,
+            test_ann_files,
+            unlabeled_data_roots,
+            unlabeled_file_list,
+            data_format,
+        )
+        self.scope = "mmseg"
+        self.dataset_registry = MMSegmentationRegistry().get("dataset")
+
+    def _get_sub_task_dataset(self) -> TorchDataset:
+        return OTXSegDataset
+
+    def _build_dataloader(
+        self,
+        dataset: TorchDataset | None,
+        batch_size: int | None = 2,
+        num_workers: int | None = 0,
+        shuffle: bool = True,
+        pin_memory: bool = False,
+        drop_last: bool = True,
+        sampler: Sampler | (Iterable | dict) | None = None,
+        persistent_workers: bool = False,
+        **kwargs,
+    ) -> TorchDataLoader | None:
+        """Builds a PyTorch DataLoader for the given dataset.
+
+        Args:
+            dataset (Optional[TorchDataset]): The dataset to load.
+            batch_size (int): The batch size to use.
+            num_workers (int): The number of worker processes to use for data loading.
+            shuffle (bool): Whether to shuffle the data.
+            pin_memory (bool): Whether to pin memory for faster GPU transfer.
+            drop_last (bool): Whether to drop the last incomplete batch.
+            sampler (Optional[Union[Sampler, Iterable, Dict]]): The sampler to use for data loading.
+            persistent_workers (bool): Whether to keep the worker processes alive between iterations.
+            **kwargs: Additional arguments to pass to the DataLoader constructor.
+
+        Returns:
+            Optional[TorchDataLoader]: The DataLoader for the given dataset.
+        """
+        # TODO (Eugene): almost identical as mmengine.runner. Need to rethink about the design.
+        # CVS-124394
+
+        if dataset is None:
+            return None
+        rank, _ = get_dist_info()
+
+        # mmengine build dataloader in runner where we build in dataset.
+        # So, I'm basically copying things from runner.build_dataloader to here
+        # which does not make sense. why do we not use dataloader_cfg and build dataloader
+        # in runner.build_dataloader?
+        seed = kwargs.get("seed", 0)
+
+        # build sampler
+        if sampler is None:
+            sampler = DATA_SAMPLERS.build(
+                {"type": "DefaultSampler", "shuffle": shuffle},
+                default_args={"dataset": dataset, "seed": seed},
+            )
+
+        init_fn = partial(worker_init_fn, num_workers=num_workers, rank=rank, seed=seed) if seed is not None else None
+        if digit_version(torch.__version__) >= digit_version("1.8.0"):
+            kwargs["persistent_workers"] = persistent_workers
```

Comment:
basically, good to have this. But aren't we going to enforce to install pytorch >= 1.13?

Reply:
I guess for now it's fine in case we wanna support 1.8.0 in the future.

Reply:
got it.